### ISSUE TITLE
Babel helper export is wrongly hoisted

### ISSUE THREAD

**Issue by TrySound** [rollup/rollup#2414]
Is rollup able to code split inlined babel helpers?

I got an issue with this. This helper is exported at the top of the chunk with `exports({ e: _extends })`

```js
var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (http://Object.prototype.hasOwnProperty.call (source, key)) { target[key] = source[key]; } } } return target; };
```

I guess it's inlined helper. Babel distributes babe-runtime with not minified helpers.

Moving `exports({})` to the bottom of the chunk should solve the problem.

 And I don't know how to reproduce this in small sandbox.

**Comment by lukastaegert** [comment]
It would be nice to have a little more information.
* Which output format does this apply to?
* In which situations does this happen?
* Do you have a repo or test case that exposes the problem? Otherwise there is no way for us to determine if a fix is working and I hate to do guess-work here.

**Comment by TrySound** [comment]
@lukastaegert @guybedford I found how to reproduce bug. Not sure it's exactly same bug though.
https://github.com/TrySound/rollup-bug-1

**Comment by TrySound** [comment]
@lukastaegert This just broke my app completely and unconditionally today. I can only remove some pages to fix it. Really annoying problem.

**Comment by TrySound** [comment]
@lukastaegert I removed all plugins except `commonjs`. Hope this will simplify your investigation.

**Comment by guybedford** [comment]
@TrySound it may help if you can share the build output, and do as much as you can to cut down the size of the build by emptying all modules that aren't affecting the issue.

**Comment by TrySound** [comment]
@guybedford Reduced a lot. Take a look if you have a time.

**Comment by TrySound** [comment]
```js
(function () {
  'use strict';

  function _extends$1() {
    _extends$1 = Object.assign || function (target) {
      for (var i = 1; i < arguments.length; i++) {
        var source = arguments[i];

        for (var key in source) {
          if (Object.prototype.hasOwnProperty.call(source, key)) {
            target[key] = source[key];
          }
        }
      }

      return target;
    };

    return _extends$1.apply(this, arguments);
  }

  var whitelist = ['en', 'fr'];
  var versions = whitelist.reduce(function (acc, lng) {
    var _extends2;

    return _extends({}, acc, (_extends2 = {}, _extends2[lng] = null, _extends2));
  }, {});

}());
```

**Comment by lukastaegert** [comment]
Ok, so this looks like `_extends` is referenced as `_extends$1` as if it has been deconflicted against itself.

**Comment by lukastaegert** [comment]
Reduced the repro as much as possible:
https://rollupjs.org/repl?version=0.66.5&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCcuJTJGZm9vMiclM0IlNUNuaW1wb3J0JTIwZm9vJTIwZnJvbSUyMCcuJTJGZm9vMSclM0IlNUNuJTVDbmNvbnNvbGUubG9nKGZvbygpKSUyMiU3RCUyQyU3QiUyMm5hbWUlMjIlM0ElMjJmb28xLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMGRlZmF1bHQlMjBmdW5jdGlvbiUyMGZvbygpJTIwJTdCJTdEJTVDbiU1Q25mb28lMjAlM0QlMjBmdW5jdGlvbigpJTdCJTdEJTNCJTIyJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMmZvbzIuanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyZnVuY3Rpb24lMjBmb28oKSU3QiU3RCUyMiU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJlc20lMjIlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE

There are actually several critical pieces to trigger this:
* There is a default export with an id that is actually used
* This id is reassigned in the same module
* There is a naming conflict that triggers this id to be deconflicted (it can also be a naming conflict with something that is eventually tree-shaken as in this example)

**Comment by lukastaegert** [comment]
Will keep investigating tomorrow.

**Comment by TrySound** [comment]
Thanks. At least I found a way to fix the issue. I just overrode extends helpers in `transform` hook.

**Comment by lukastaegert** [comment]
Potential fix at #2502. Could you check if this also fixes the original issue?

**Comment by TrySound** [comment]
Just tried. Works perfectly. Thank you
